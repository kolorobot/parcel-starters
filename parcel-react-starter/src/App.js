import React, { useEffect, useState } from 'react';

export default function App() {

    const [count, setCount] = useState(0);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        setIsLoading(true);
        setTimeout(() => {
            setCount(1);
            setIsLoading(false);
        }, Math.floor(Math.random() * 1000));
    }, []);

    return (
        <button data-testid='count-button' disabled={isLoading} onClick={() => setCount(count + 1)}>{count}</button>
    );
}
