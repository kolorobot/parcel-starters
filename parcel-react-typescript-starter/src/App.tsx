import * as React from 'react';
import { useEffect, useState } from 'react';

type AppProps = {
    onCounterUpdate?(value: number): void;
}

const App: React.FC<AppProps> = ({ onCounterUpdate }) => {

    const [count, setCount] = useState(0);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        setIsLoading(true);
        setTimeout(() => {
            setCount(1);
            setIsLoading(false);
        }, Math.floor(Math.random() * 1000));
    }, []);

    const handleClick = () => {
        setCount(count + 1);
        if (onCounterUpdate) {
            onCounterUpdate(count);
        }
    };

    return (
        <button data-testid='count-button' disabled={isLoading} onClick={handleClick}>{count}</button>
    );
};

export default App;
