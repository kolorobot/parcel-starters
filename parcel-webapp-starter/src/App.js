export default function App() {
    const div = document.createElement('div');
    div.innerHTML = `
    <label for='text'>Enter some text</label>
    <input id='text' />
    <button>Print</button>`;

    const button = div.querySelector('button');
    const input = div.querySelector('input');

    button.addEventListener('click', () => {
        setTimeout(() => {
            const resultContainer = document.createElement('div');
            resultContainer.innerHTML = `<div data-testid='printed-text'>${input.value}</div>`;
            div.appendChild(resultContainer);
        }, Math.floor(Math.random() * 200));
    });
    return div;
}
