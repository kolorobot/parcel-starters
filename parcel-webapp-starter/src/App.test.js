import App from './App';
import {
    getByLabelText,
    getByTestId,
    getByText,
    queryByTestId,
    waitFor,
} from '@testing-library/dom';

describe('App', () => {
    it('prints the provided text', async () => {
        const text = 'Lorem ipsum';
        const container = App();

        const input = getByLabelText(container, 'Enter some text');
        input.value = text;

        getByText(container, 'Print').click();

        await waitFor(() => expect(
            queryByTestId(container, 'printed-text')).toBeTruthy(),
        );

        expect(getByTestId(container, 'printed-text')).toHaveTextContent(text);

        expect(container).toMatchSnapshot();
    });
});
