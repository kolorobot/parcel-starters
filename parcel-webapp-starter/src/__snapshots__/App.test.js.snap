// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing It prints the provided text 1`] = `
<div>
  
    
  <label
    for="text"
  >
    Enter some text
  </label>
  
    
  <input
    id="text"
  />
  
    
  <button>
    Print
  </button>
  <div>
    <div
      data-testid="printed-text"
    >
      Lorem ipsum
    </div>
  </div>
</div>
`;

exports[`App testing prints the provided text 1`] = `
<div>
  
    
  <label
    for="text"
  >
    Enter some text
  </label>
  
    
  <input
    id="text"
  />
  
    
  <button>
    Print
  </button>
  <div>
    <div
      data-testid="printed-text"
    >
      Lorem ipsum
    </div>
  </div>
</div>
`;
